realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -67,6 +67,17 @@
                 srcTableType);
     }
 
+    static RealmResults<DynamicRealmObject> createBacklinkResults(DynamicRealm realm, Row row, Table srcTable, String srcFieldName) {
+        if (!(row instanceof UncheckedRow)) {
```

Comment:
If this method return a dynamic results always, then the `row` should only be `CheckedRow`. Also, the comments at https://github.com/realm/realm-java/blob/master/realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java#L72 is not correct :P